### DISCUSSION TITLE
How to gain best possible quality, disregard performance.

### DISCUSSION THREAD

**Discussion by JFFby** [saucecontrol/PhotoSauce#148] [Q&A] [answered]
Hey guys,

I want to express my gratitude for this library, and I really enjoy using it. However, as a beginner in raster graphics, I am curious if there is a configuration that prioritizes image quality over performance (I don't care about performance at all). I think my question may be common, and it would be helpful to include the answer in the documentation.

Thank you in advance!

**Comment by saucecontrol** [comment, marked as answer]
Howdy!  Thanks for the kind words.

Quality is subjective.  At the end of the day, if you are resizing an image, you are *guessing* what the image would have looked like if were captured at a different size.  That guess can only be worse than that hypothetical capture, so "best quality" can mean different things.  It could mean most accurate, or it could mean most visually pleasing, or any number of other priorities.

Most people, most of the time, on most image sources, will prefer a sharper image.  This is the default that you get with MagicScaler, and it happens to be heavily optimized for that case, so it's also very fast at very high 'quality'.

Some images, some of the time, will look better without sharpening or with a different resampling algorithm.  ImageMagick (no relation) has [extensive documentation](https://imagemagick.org/Usage/filter/) on common resampling filters, most of which are implemented (or definable) in MagicScaler.

Unless you happen to have an image that doesn't look good with the default settings, the only recommendation I can make is to stick with the defaults.  The options that are available are for users who have specific requirements, but there's no need to fiddle with them in search of ultimate quality.  There's really no such thing.

All that said, it *is* possible to get the algorithms wrong, and unfortunately many of the image manipulation libraries in existence do get things wrong in one way or another (bad math, bad defaults, bad implementations, etc).  The comments about quality in this library's documentation mostly refer to those things.
